src/navier.f90

```diff
@@ -1280,5 +1289,246 @@ subroutine tbl_flrt (ux1,uy1,uz1)
     enddo
 
   end subroutine tbl_flrt
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !!
+  !!  subroutine: pipe_bulk / pipe_bulk_u / pipe_bulk_phi
+  !!      AUTHOR: Rodrigo Vicente Cruz
+  !! DESCRIPTION: Correction of pipe's bulk velocity (constant 
+  !!              flow rate) and bulk temperature.
+  !!              See Thesis Vicente Cruz 2021 for help.
+  !!
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !********************************************************************
+  !
+  subroutine pipe_bulk(ux,uy,uz,ep)
+  !
+  !********************************************************************
+
+    use param
+    use decomp_2d
+    use variables
+    use var,  only: phi1
+
+    implicit none
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3))  :: ux,uy,uz,ep
+    integer                                             :: is
+
+    !Bulk velocity correction
+    call pipe_bulk_u(ux,uy,uz,ep,one)
+
+    !Bulk temperature correction
+    if (numscalar.ne.0) then
+        do is=1,numscalar
+            call pipe_bulk_phi(phi1(:,:,:,is),ux,ep,is,one)
+        enddo
+    endif
+
+  end subroutine pipe_bulk
+  !********************************************************************
+  !
+  subroutine pipe_bulk_u(ux,uy,uz,ep,ub_constant)
+  !
+  !********************************************************************
+
+    use decomp_2d
+    use variables
+    use param
+    use var
+    use ibm_param, only: rai
+    use MPI
+
+    implicit none
+    !INPUTS
+    real(mytype),intent(inout),dimension(xsize(1),xsize(2),xsize(3))    :: ux,uy,uz
+    real(mytype),intent(in   ),dimension(xsize(1),xsize(2),xsize(3))    :: ep
+    real(mytype),intent(in   )                                          :: ub_constant !bulk velocity value
+    !LOCALS
+    real(mytype)                                                        :: qm      !flow rate
+    real(mytype)                                                        :: ym,zm,yc,zc,r
+    real(mytype)                                                        :: ncount
```

Comment:
Unused variable ?

Reply:
The unused variables have been removed in 25d46cc43400701dbc33aadb5d8c25638838e334.